src/Compilers/VisualBasic/Portable/Symbols/Tuples/TupleEventSymbol.vb

```diff
@@ -0,0 +1,106 @@
+            Get
+                Return Me._containingType
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property Type() As TypeSymbol
+            Get
+                Return Me._underlyingEvent.Type
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property AddMethod() As MethodSymbol
+            Get
+                Return Me._containingType.GetTupleMemberSymbolForUnderlyingMember(Of MethodSymbol)(Me._underlyingEvent.AddMethod)
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property RemoveMethod() As MethodSymbol
+            Get
+                Return Me._containingType.GetTupleMemberSymbolForUnderlyingMember(Of MethodSymbol)(Me._underlyingEvent.RemoveMethod)
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property AssociatedField() As FieldSymbol
+            Get
+                Return Me._containingType.GetTupleMemberSymbolForUnderlyingMember(Of FieldSymbol)(Me._underlyingEvent.AssociatedField)
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property RaiseMethod As MethodSymbol
+            Get
+                Return Me._containingType.GetTupleMemberSymbolForUnderlyingMember(Of MethodSymbol)(Me._underlyingEvent.RaiseMethod)
+            End Get
+        End Property
+
+        Friend Overrides ReadOnly Property IsExplicitInterfaceImplementation() As Boolean
+            Get
+                Return Me._underlyingEvent.IsExplicitInterfaceImplementation
+            End Get
+        End Property
+
+        Public Overrides ReadOnly Property ExplicitInterfaceImplementations() As ImmutableArray(Of EventSymbol)
+            Get
+                Return Me._underlyingEvent.ExplicitInterfaceImplementations
+            End Get
+        End Property
+
+        Public Sub New(container As TupleTypeSymbol, underlyingEvent As EventSymbol)
+            MyBase.New(underlyingEvent)
+            Me._containingType = container
+        End Sub
+
+        Friend Overrides Function GetUseSiteErrorInfo() As DiagnosticInfo
+            Dim useSiteDiagnostic As DiagnosticInfo = MyBase.GetUseSiteErrorInfo
+            MyBase.MergeUseSiteErrorInfo(useSiteDiagnostic, Me._underlyingEvent.GetUseSiteErrorInfo())
+            Return useSiteDiagnostic
+        End Function
+
+        Public Overrides Function GetHashCode() As Integer
+            Return Me._underlyingEvent.GetHashCode()
+        End Function
+
+        Public Overrides Function Equals(obj As Object) As Boolean
+            Return Me.Equals(TryCast(obj, TupleEventSymbol))
+        End Function
+
+        Public Overloads Function Equals(other As TupleEventSymbol) As Boolean
+            Dim flag As Boolean = other Is Me
+            Return flag OrElse (other IsNot Nothing AndAlso Me._containingType Is other._containingType AndAlso Me._underlyingEvent Is other._underlyingEvent)
+        End Function
```

Comment:
Where are we using deep `Equals` for these members? And if we do expect multiple instances of the same member, shouldn't `Equals` compare `_containingType` with `Equals` rather than `Is`?